src/qt/window.cpp

```diff
@@ -1517,17 +1517,23 @@ bool wxWindowQt::QtHandleKeyEvent ( QWidget *WXUNUSED( handler ), QKeyEvent *eve
         }
 #endif // wxUSE_ACCEL
 
-        e.SetEventType( wxEVT_CHAR );
-        e.SetEventObject(this);
+        // For compatibility with wxWidgets, don't generate wxEVT_CHAR event for
+        // the following keys: SHIFT, CONTROL, MENU, CAPITAL, NUMLOCK and SCROLL.
+        const int key = event->key();
+        if ( key < Qt::Key_Shift || key > Qt::Key_ScrollLock )
```

Comment:
This is a very non-obvious way of testing for what the comment above says. It would be much clearer to just write

```cpp
switch ( event->key() )
{
    case Qt::Key_Shift:
    ...
    case Qt::Key_ScrollLock:
        // Skip event generation.
        return handled;
}
```